Software/Human-Zombie-Device/Human-Zombie-Device.ino

```diff
@@ -18,58 +21,29 @@ void setup() {
     write_oled_line(1, "ZTAG DEVICE");
     Serial.println("ZTAG DEVICE");
 
-    if (!init_lora()) {
-        Serial.println("Starting LoRa failed!");
-        write_oled_and_serial_line(3, "!!LORA FAILED!!");
-        while (1)
-            ;
-    }
-    Serial.println("LoRa Initializing OK!");
+    bleManager.initBleZTagDevice();
+    Serial.println("Ble Initializing");
 
     // Get the saved state from EEPROM and output it
     deviceState = init_and_read_device_state();
     numberOfInfectionTicks = 0;   // set the infection counter to zero.
     write_oled_and_serial_line(2, DeviceStateToString(deviceState));
+
+    bleManager.updateZtagBeaconName(DeviceStateToString(DeviceState::human));
+    
 }
 
 void loop() {
 
-    if (is_message_ztag(get_new_lora_packet())) {
-
-        DeviceState receivedState = StringToDeviceState(receivedPacket.message);
-        // Main behaviour loop, only entered if we find a ZTag message
-        if (DeviceState::human == deviceState) {
-
-            if (DeviceState::infected == receivedState) {
-                is_infected_yet(receivedPacket.rssi);
-            }
-
-            // TODO: Decision time, currently we don't reset the health bar if
-            // you go in a medical
-            //  zone whilst a human. Maybe we should. If we should then set to
-            //  medicalzone here.
-
-        } else if (DeviceState::infected == deviceState) {
-
-            if (DeviceState::medicalzone == receivedState) {
-                deviceState = DeviceState::medicalzone;
-            }
-
-        } else if (DeviceState::medicalzone == deviceState) {
-
-            Serial.println("Currently in a medical zone");
-            reset_infection_ticker();
-            reset_health();
-            deviceState = DeviceState::human;
-        }
-    } else {
-        Serial.println("Packet something else ignoring");
+    bleManager.pollBle(); 
+    
+    if (bleManager.getFoundZtagDevices().size() > 0) {
+        // If we find a zTag device then store that in foundZTagData        
+        Serial << "number of ZTags found: " << bleManager.getFoundZtagDevices().size() << '\n';
```

Comment:
Should be using endl instead of \n ? 

I know on an embedded micro it matters very little as there is no buffer to flush, but I would have expected it to be endl, because C++ ?

Reply:
Hmm, This may be not possible with how you have done the printf in [SerialHelper.h](https://github.com/gppk/ZombieTag/pull/22/files#diff-b854b58fe09a9b95d010b9ea59c010ca45a7fec88ca8cf6540cb59b578cb812f)

Interested to see if it is

Reply:
No. because im not using the std library just wrapping up a print()